File: packages/modules/lodash/core/go/core/math/math_test.go
package math

import (
	"testing"
)

func TestAdd(t *testing.T) {
	if got := Add(6, 4); got != 10 {
		t.Errorf("expected 10, got %f", got)
	}
	if got := Add(-2, 3); got != 1 {
		t.Errorf("expected 1, got %f", got)
	}
}

func TestCeil(t *testing.T) {
	if got := Ceil(4.006); got != 5 {
		t.Errorf("expected 5, got %f", got)
	}
	if got := Ceil(4.0); got != 4 {
		t.Errorf("expected 4, got %f", got)
	}
	if got := Ceil(-4.5); got != -4 {
		t.Errorf("expected -4, got %f", got)
	}
}

func TestDivide(t *testing.T) {
	if got := Divide(6, 4); got != 1.5 {
		t.Errorf("expected 1.5, got %f", got)
	}
	if got := Divide(0, 5); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
	if got := Divide(10, 3); got != 10.0/3.0 {
		t.Errorf("expected %f, got %f", 10.0/3.0, got)
	}
}

func TestFloor(t *testing.T) {
	if got := Floor(4.006); got != 4 {
		t.Errorf("expected 4, got %f", got)
	}
	if got := Floor(4.0); got != 4 {
		t.Errorf("expected 4, got %f", got)
	}
	if got := Floor(-4.5); got != -5 {
		t.Errorf("expected -5, got %f", got)
	}
}

func TestMax(t *testing.T) {
	if got := Max([]float64{4, 2, 8, 6}); got != 8 {
		t.Errorf("expected 8, got %f", got)
	}
	if got := Max([]float64{-1, -5, -3}); got != -1 {
		t.Errorf("expected -1, got %f", got)
	}
}

func TestMean(t *testing.T) {
	if got := Mean([]float64{4, 2, 8, 6}); got != 5 {
		t.Errorf("expected 5, got %f", got)
	}
	if got := Mean([]float64{1, 2, 3}); got != 2 {
		t.Errorf("expected 2, got %f", got)
	}
	if got := Mean([]float64{}); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
}

func TestMin(t *testing.T) {
	if got := Min([]float64{4, 2, 8, 6}); got != 2 {
		t.Errorf("expected 2, got %f", got)
	}
	if got := Min([]float64{-1, -5, -3}); got != -5 {
		t.Errorf("expected -5, got %f", got)
	}
}

func TestMultiply(t *testing.T) {
	if got := Multiply(6, 4); got != 24 {
		t.Errorf("expected 24, got %f", got)
	}
	if got := Multiply(-2, 3); got != -6 {
		t.Errorf("expected -6, got %f", got)
	}
	if got := Multiply(0, 5); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
}

func TestRound(t *testing.T) {
	if got := Round(4.1); got != 4 {
		t.Errorf("expected 4, got %f", got)
	}
	if got := Round(4.9); got != 5 {
		t.Errorf("expected 5, got %f", got)
	}
	if got := Round(4.5); got != 5 {
		t.Errorf("expected 5, got %f", got)
	}
	if got := Round(-4.5); got != -5 {
		t.Errorf("expected -5, got %f", got)
	}
}

func TestSubtract(t *testing.T) {
	if got := Subtract(6, 4); got != 2 {
		t.Errorf("expected 2, got %f", got)
	}
	if got := Subtract(1, 5); got != -4 {
		t.Errorf("expected -4, got %f", got)
	}
}

func TestSum(t *testing.T) {
	if got := Sum([]float64{4, 2, 8, 6}); got != 20 {
		t.Errorf("expected 20, got %f", got)
	}
	if got := Sum([]float64{}); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
	if got := Sum([]float64{-1, 1}); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
}

func TestMaxBy(t *testing.T) {
	objects := []map[string]float64{
		{"n": 1},
		{"n": 5},
		{"n": 3},
	}
	iteratee := func(obj map[string]float64) float64 {
		return obj["n"]
	}
	if got := MaxBy(objects, iteratee); got != 5 {
		t.Errorf("expected 5, got %f", got)
	}

	if got := MaxBy([]map[string]float64{}, iteratee); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
}

func TestMeanBy(t *testing.T) {
	objects := []map[string]float64{
		{"n": 1},
		{"n": 2},
		{"n": 3},
	}
	iteratee := func(obj map[string]float64) float64 {
		return obj["n"]
	}
	if got := MeanBy(objects, iteratee); got != 2 {
		t.Errorf("expected 2, got %f", got)
	}

	if got := MeanBy([]map[string]float64{}, iteratee); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
}

func TestMinBy(t *testing.T) {
	objects := []map[string]float64{
		{"n": 5},
		{"n": 1},
		{"n": 3},
	}
	iteratee := func(obj map[string]float64) float64 {
		return obj["n"]
	}
	if got := MinBy(objects, iteratee); got != 1 {
		t.Errorf("expected 1, got %f", got)
	}

	if got := MinBy([]map[string]float64{}, iteratee); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
}

func TestSumBy(t *testing.T) {
	objects := []map[string]float64{
		{"n": 1},
		{"n": 2},
		{"n": 3},
	}
	iteratee := func(obj map[string]float64) float64 {
		return obj["n"]
	}
	if got := SumBy(objects, iteratee); got != 6 {
		t.Errorf("expected 6, got %f", got)
	}

	if got := SumBy([]map[string]float64{}, iteratee); got != 0 {
		t.Errorf("expected 0, got %f", got)
	}
}

File: packages/modules/lodash/core/typescript/tests/math.test.ts
import {
  add,
  ceil,
  divide,
  floor,
  max,
  maxBy,
  mean,
  min,
  minBy,
  multiply,
  round,
  subtract,
  sum,
  sumBy,
} from '../src/math.js';

describe('add', () => {
  it('should add', () => {
    expect(add(6, 4)).toBe(10);
  });
});

describe('ceil', () => {
  it('should ceil', () => {
    expect(ceil(4.006)).toBe(5);
    expect(ceil(6.004, 2)).toBe(6.01);
    expect(ceil(6040, -2)).toBe(6100);
  });
});

describe('divide', () => {
  it('should divide', () => {
    expect(divide(6, 4)).toBe(1.5);
  });
});

describe('floor', () => {
  it('should floor', () => {
    expect(floor(4.006)).toBe(4);
    expect(floor(6.004, 2)).toBe(6);
  });
});

describe('max', () => {
  it('should find max', () => {
    expect(max([4, 2, 8, 6])).toBe(8);
    expect(max([])).toBeUndefined();
  });
});

describe('maxBy', () => {
  it('should find max by iteratee', () => {
    expect(maxBy([{ n: 1 }, { n: 2 }], (o) => o.n)).toEqual({ n: 2 });
  });
});

describe('mean', () => {
  it('should compute mean', () => {
    expect(mean([4, 2, 8, 6])).toBe(5);
    expect(mean([])).toBeNaN();
  });
});

describe('min', () => {
  it('should find min', () => {
    expect(min([4, 2, 8, 6])).toBe(2);
    expect(min([])).toBeUndefined();
  });
});

describe('minBy', () => {
  it('should find min by iteratee', () => {
    expect(minBy([{ n: 1 }, { n: 2 }], (o) => o.n)).toEqual({ n: 1 });
  });
});

describe('multiply', () => {
  it('should multiply', () => {
    expect(multiply(6, 4)).toBe(24);
  });
});

describe('round', () => {
  it('should round', () => {
    expect(round(4.006)).toBe(4);
    expect(round(4.006, 2)).toBe(4.01);
  });
});

describe('subtract', () => {
  it('should subtract', () => {
    expect(subtract(6, 4)).toBe(2);
  });
});

describe('sum', () => {
  it('should sum', () => {
    expect(sum([4, 2, 8, 6])).toBe(20);
  });
});

describe('sumBy', () => {
  it('should sum by iteratee', () => {
    expect(sumBy([{ n: 4 }, { n: 2 }, { n: 8 }, { n: 6 }], (o) => o.n)).toBe(
      20
    );
  });
});
